src/ext/oblivc/obliv_bits.c

```diff
@@ -402,26 +402,32 @@ void setCurrentParty(ProtocolDesc* pd, int party)
 ProtocolDesc* ocCurrentProto() { return currentProto; }
 void ocSetCurrentProto(ProtocolDesc* pd) { currentProto=pd; }
 
-void ocSplitProto(ProtocolDesc* pdout, ProtocolDesc * pdin)
-{
-  *pdout = (struct ProtocolDesc) {
-    .error = pdin->error,
-    .partyCount = pdin->error,
-    .currentParty = pdin->currentParty,
-    .feedOblivInputs = pdin->feedOblivInputs,
-    .revealOblivBits = pdin->revealOblivBits,
-    .setBitAnd = pdin->setBitAnd,
-    .setBitOr = pdin->setBitOr,
-    .setBitXor = pdin->setBitXor,
-    .setBitNot = pdin->setBitNot,
-    .flipBit = pdin->flipBit,
-    .thisParty = pdin->thisParty,
-    .trans = pdin->trans->split(pdin->trans),
-    .splitextra = pdin->splitextra,
-    .cleanextra = pdin->cleanextra,
-    .extra = NULL
-  };
-  if (pdout->splitextra != NULL && pdin->extra != NULL) pdout->splitextra(pdout, pdin);
+bool ocCanSplitProto(ProtocolDesc * pdin) { return (pdin->trans->split != NULL && pdin->splitextra != NULL && pdin->extra != NULL); }
```

Comment:
Please keep lines to 80 chars, not least so I can actually see the whole text :)